core/replica.go

```diff
@@ -96,7 +105,13 @@ func (r *Replica) Start() error {
 		// connected.
 		_, err = sh.HandleMessageStream(out)
 		if err != nil {
-			return fmt.Errorf("Error establishing connection to peer replica %d: %s", i, err)
+			f++
+			if f <= r.f {
+				fmt.Printf("Error establishing connection to peer replica %d: %s\n", i, err)
```

Comment:
Do a caller have a way to get how many/which nodes it can connect successfully?

Reply:
Probably we can have it for example with `sync.WaitGroup`, I'l try it later.